dask/array/core.py

```diff
@@ -669,9 +669,11 @@ def map_blocks(func, *args, **kwargs):
         raise TypeError(msg % type(func).__name__)
     name = kwargs.pop('name', None)
     token = kwargs.pop('token', None)
-    if not name:
-        name = '%s-%s' % (token or funcname(func),
-                          tokenize(token or func, args, **kwargs))
+    if token:
+        warnings.warn("The token= keyword to map_blocks has been moved to name=")
```

Comment:
Should we note when we plan to drop `token`?

Reply:
Perhaps?  When I run into warnings like this the future version number rarely means much to me.  I mostly care about "how far into the future will this work?"  the answer I get is some version number, which doesn't mean much to me, so I go ahead and change my code anyway.

Happy to adhere to some convention though if it exists.

Reply:
Any further concern here @jakirkham ?

Reply:
This point of including versions in deprecations came up once in a NumPy PR I submitted. So it seemed like a fair point to raise. That said, I don't have particularly strong feelings about it.

Reply:
OK, in that case I'll probably pass on this.